api/api/src/main/scala/uk/ac/wellcome/platform/api/services/CollectionService.scala

```diff
@@ -0,0 +1,65 @@
+package uk.ac.wellcome.platform.api.services
+
+import scala.concurrent.{ExecutionContext, Future}
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.{ElasticClient, ElasticError, Index}
+import com.sksamuel.elastic4s.requests.searches.{SearchHit, SearchResponse}
+
+import uk.ac.wellcome.models.work.internal._
+import uk.ac.wellcome.models.work.internal.result._
+import uk.ac.wellcome.models.work.internal.CollectionTree
+import uk.ac.wellcome.models.Implicits._
+import uk.ac.wellcome.json.JsonUtil.fromJson
+import uk.ac.wellcome.platform.api.Tracing
+
+/**
+  * Retrieves a CollectionTree given some paths to expand.
+  *
+  * For example, if asked to expand the path "A/B", it will return a tree
+  * consisting of the node A, the child node B, and any children of either of
+  * those nodes.
+  *
+  * A
+  * |______
+  * |  |  |
+  * B  ?  ?
+  * |______
+  * |  |  |
+  * ?  ?  ?
```

Comment:
I have those Unicode characters to hand, which you're welcome to use:

```suggestion
  * A
  * ├── B
  * │   ├── ?
  * │   ├── ?
  * │   └── ?
  * │
  * ├── ?
  * └── ?
```